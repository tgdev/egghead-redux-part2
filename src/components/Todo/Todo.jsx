import React from 'react';
import PropTypes from 'prop-types';

const Todo = ({ text, completed, handleTodo }) => {
  return (
    <li
      onClick={handleTodo}
      style={{
        textDecoration: completed ? 'line-through' : 'none'
      }} >
      {text}
    </li>
  );
};

Todo.propTypes = {
  text: PropTypes.string.isRequired,
  completed: PropTypes.bool.isRequired,
  handleTodo: PropTypes.func.isRequired
};

export default Todo;
