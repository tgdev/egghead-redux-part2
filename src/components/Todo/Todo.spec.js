import React from 'react';
import { shallow, mount } from 'enzyme';

import Todo from './Todo';

describe('Todo component', () => {
  let mockClickHandler = jest.fn();

  afterEach(() => {
    mockClickHandler.mockReset();
  });

  it('renders Todo correctly', () => {
    const wrapper = shallow(<Todo text='Learn redux' completed={false} handleTodo={mockClickHandler} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders a completed todo with line-through', () => {
    const wrapper = shallow(<Todo text='Learn redux' completed={true} handleTodo={mockClickHandler} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('triggers handleToggleTodo when todo item is clicked', () => {
    const wrapper = mount(<Todo text='Learn redux' completed={true} handleTodo={mockClickHandler} />);
    wrapper.find('li').first().simulate('click');
    expect(mockClickHandler).toHaveBeenCalledTimes(1);
  });
});
