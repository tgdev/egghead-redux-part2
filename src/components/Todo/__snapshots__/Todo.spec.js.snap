// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo component renders Todo correctly 1`] = `
<li
  onClick={[Function]}
  style={
    Object {
      "textDecoration": "none",
    }
  }
>
  Learn redux
</li>
`;

exports[`Todo component renders a completed todo with line-through 1`] = `
<li
  onClick={[Function]}
  style={
    Object {
      "textDecoration": "line-through",
    }
  }
>
  Learn redux
</li>
`;
